airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/manifest.yaml

```diff
@@ -370,10 +373,118 @@ definitions:
+      paginator:
+        $ref: "#/definitions/links_next_paginator"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - type: ParentStreamConfig
+            parent_key: id
+            partition_field: post_id
+            stream:
+              $ref: "#/definitions/posts_stream"
+            incremental_dependency: true
+      record_selector:
+        extractor:
+          type: DpathExtractor
+          field_path: ["comments"]
+    transformations:
+      - type: AddFields
+        fields:
+          - path:
+              - _airbyte_parent_id # used for post_comment_votes stream
+            value: "{{ { 'post_id': record['post_id'], 'comment_id': record['id'] } }}"
+
+  post_votes_stream:
+    $ref: "#/definitions/base_incremental_stream"
+    name: "post_votes"
+    primary_key: "id"
+    schema_loader:
+      type: JsonFileSchemaLoader
+      file_path: "./source_zendesk_support/schemas/votes.json"
+    incremental_sync:
+      $ref: "#/definitions/cursor_incremental_sync"
+      start_time_option:
+        $ref: "#/definitions/cursor_incremental_sync/start_time_option"
+        field_name: "start_time"
+    retriever:
+      $ref: "#/definitions/retriever"
+      ignore_stream_slicer_parameters_on_paginated_requests: true
+      requester:
+        $ref: "#/definitions/retriever/requester"
+        path: "community/posts/{{ stream_partition.post_id }}/votes"
+      paginator:
+        $ref: "#/definitions/links_next_paginator"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - type: ParentStreamConfig
+            parent_key: id
+            partition_field: post_id
+            stream:
+              $ref: "#/definitions/posts_stream"
+            incremental_dependency: true
+      record_selector:
+        extractor:
+          type: DpathExtractor
+          field_path: ["votes"]
+
+  post_comment_votes_stream:
+    $ref: "#/definitions/base_incremental_stream"
+    name: "post_comment_votes"
+    primary_key: "id"
+    schema_loader:
+      type: JsonFileSchemaLoader
+      file_path: "./source_zendesk_support/schemas/votes.json"
+    incremental_sync:
+      $ref: "#/definitions/cursor_incremental_sync"
+      start_time_option:
+        $ref: "#/definitions/cursor_incremental_sync/start_time_option"
+        field_name: "start_time"
+    retriever:
+      $ref: "#/definitions/retriever"
+      ignore_stream_slicer_parameters_on_paginated_requests: true
+      requester:
+        $ref: "#/definitions/retriever/requester"
+        path: "community/posts/{{ stream_partition.id.post_id }}/comments/{{ stream_partition.id.comment_id }}/votes"
+      paginator:
+        $ref: "#/definitions/links_next_paginator"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - type: ParentStreamConfig
+            parent_key: _airbyte_parent_id
```

Comment:
just so i understand this right, this part of the code will take the existing record's `_airbyte_parent_id` value and make it accessible via the `stream_partition` under `id` which is how we access and use it in the requester `path` above?

Reply:
Exactly! We needed to add the transformation in `post_comments` so that we could have a composite key that we can extract and re-use in `post_comment_votes`. This is used in the path as `path: "community/posts/{{ stream_partition.id.post_id }}/comments/{{ stream_partition.id.comment_id }}/votes"`